src/figure/trader.c

```diff
@@ -68,9 +69,21 @@ int trader_has_traded(int trader_id)
 
 int trader_has_traded_max(int trader_id)
 {
-    return data.traders[trader_id].bought_amount >= 12 || data.traders[trader_id].sold_amount >= 12;
+    return data.traders[trader_id].bought_amount >= figure_trade_sea_trade_units() || data.traders[trader_id].sold_amount >= figure_trade_sea_trade_units();
```

Comment:
Like!!! :+1: The fewer magic numbers the better.

But you can go even further:

`return trader_has_bought_max(trader_id) || trader_has_sold_max(trader_id);`

This will likely be inlined in compiler optimizations.

Reply:
fixed